tensorflow/go/operation_test.go

```diff
@@ -166,6 +166,71 @@ func TestOutputDataTypeAndShape(t *testing.T) {
 	}
 }
 
+func TestOperationInputs(t *testing.T) {
+	g := NewGraph()
+	x, err := Placeholder(g, "x", Float)
+	if err != nil {
+		t.Fatal(err)
+	}
+	y, err := Placeholder(g, "y", Float)
+	if err != nil {
+		t.Fatal(err)
+	}
+	outputs := []Output{x, y}
+	add, err := Add(g, "add", x, y)
+	if err != nil {
+		t.Fatal(err)
+	}
+	addOp := add.Op
+
+	if out := addOp.NumInputs(); out != 2 {
+		t.Fatalf("Got %d inputs, wanted 2", out)
+	}
+
+	if out := addOp.Input(0).DataType(); out != Float {
+		t.Fatalf("Got %v type, wanted %v", out, Float)
+	}
+
+	for i := 0; i < addOp.NumInputs(); i++ {
+		input := addOp.Input(i)
+		got := input.Input().Op.Name()
+		want := outputs[i].Op.Name()
+		if want != got {
+			t.Fatalf("Got op name %q, wanted %q", got, want)
+		}
+	}
+}
+
+func TestOperationConsumers(t *testing.T) {
+	g := NewGraph()
+	x, err := Placeholder(g, "x", Float)
+	if err != nil {
+		t.Fatal(err)
+	}
+	a, err := Neg(g, "a", x)
+	if err != nil {
+		t.Fatal(err)
+	}
+	b, err := Neg(g, "b", x)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	consumers := []*Operation{a.Op, b.Op}
+
+	if out := x.NumConsumers(); out != 2 {
+		t.Fatalf("Got %d consumers, wanted 2", out)
+	}
+
+	for i, consumer := range x.Consumers() {
```

Comment:
Shouldn't we check that `len(x.Consumers())` is 2 as well? Otherwise this test would pass even with an implemention of `return nil` for `Consumers()`?

Reply:
Removed NumConsumers and now check for len(x.Consumers())